client/components/stats-date-control/index.tsx

```diff
@@ -80,12 +118,18 @@ const StatsDateControl = ( {
 		const endDate = anchor.format( 'YYYY-MM-DD' );
 		const startDate = anchor.subtract( shortcut.range, 'days' ).format( 'YYYY-MM-DD' );
 
-		recordTracksEvent( `${ event_from }_stats_date_picker_shortcut_${ shortcut.id }_clicked` );
+		recordTracksEvent( eventNames[ event_from ][ shortcut.id as EventNameKey ] );
```

Comment:
If we could set shortcut.id the same type as EventNameKey, then the error above would be caught before shipping 😄 

https://github.com/Automattic/wp-calypso/pull/94834/files#r1776037198